twill-api/src/main/java/org/apache/twill/api/TwillRunResources.java

```diff
@@ -59,6 +61,16 @@
   /**
    * @return the enabled log level for the container where the runnable is running in.
    */
-  LogEntry.Level getLogLevel();
+  LogEntry.Level getRootLogLevel();
+
+  /**
+   * @return the enabled log level arguments for the container where the runnable is running in.
+   */
+  Map<String, LogEntry.Level> getLogLevelArguments();
 
+  /**
+   * Update the log level arguments for the container where the runnable is running in.
+   * @param logLevelArguments
+   */
+  void updateLogLevel(Map<String, LogEntry.Level> logLevelArguments);
```

Comment:
Why there is an update method? This interface is supposedly for reporting, hence is immutable.